File: course/exercise_steady_state_filter/plot_obs_ens.m
% Ensemble Kalman filter:
% Make plot to show difference between observations and predicted values
%
% author: Nils van Velzen
%
[t, ens, obs, pred_f, pred_a]=load_ensemble('enkf_results');

diff=pred_a-obs;
figure(1);
plot(diff')
title('Difference between model and observations');
ylabel('prediction-observation');
xlabel('time');

[nobs,ntimes]=size(diff)
for iobs=1:nobs
   rms=norm(diff(iobs,:))/sqrt(ntimes);
   disp(['RMS of location ',num2str(iobs),' ',num2str(rms)]);
end

File: course/exercise_steady_state_filter/plot_obs_ens.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
"""
Simulation of model:
Make plot to show difference between observations and predicted values

@author: verlaanm
"""
#load numpy and matplotlib if needed
import numpy as np
import matplotlib.pyplot as plt

import enkf_results as enkf

diff = enkf.obs -enkf.pred_a_central

#
#plot residual
#
fig2 = plt.figure()
plt.plot(enkf.analysis_time,diff)
plt.title("Difference between EnKF and observations")
plt.xlabel("time")
plt.ylabel("prediction-observation")
plt.savefig("figure_2.png")

#
# Compute RMS of errors
#
(ntimes,nobs)=np.shape(diff)

rms_enkf=[]
print("===================================================")
print("RMS errors at observations locations after analysis")
print("Ensemble Kalman filter")
print("===================================================")
for iobs in range(nobs):
    rms_enkf.append(  np.sqrt(np.sum(np.square(diff[:,iobs])))  )
    print("  "+str(iobs)+" : "+str(rms_enkf[iobs]))

